### PR TITLE
Add a Git repository to Config Server via CLI

### PR DESCRIPTION
**Pull request by spikymonkey**
* Adds `config-server-add-git` CLI command
* Refactor repo manipulation command tests to share examples

I think over time we'll adapt this to support the rest of the
configuration parameters that can be supplied when specifying a git repo
to add. But for now, the story asks for support of a command of the form

```
cf config-server-add-git config-server URI
```

so this is what I've gone for. It might be that we want to support this
simple URI style to allow quick adds, and a more verbose JSON blob for
more involved configurations.

connected to pivotal-cf/spring-cloud-service-broker#320

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -87,9 +87,16 @@ func (c *Plugin) Run(cliConnection plugin.CliConnection, args []string) {
 			return config.Encrypt(cliConnection, configServerInstanceName, plainText, fileToEncrypt, authClient)
 		})
 
+	case "config-server-add-git":
```
Comment (georgeharley) [inline plugin.go]: Also need to register the `config-server-add-git` command in the `GetMetadata` function near the bottom of the file.

File: plugin.go [hunk 2]
```diff
@@ -167,6 +174,11 @@ func (c *Plugin) Run(cliConnection plugin.CliConnection, args []string) {
 		os.Exit(0) // Ignore CLI-MESSAGE-UNINSTALL etc.
 	}
 }
+
+func getGitRepoConfiguration(ac *cli.ArgConsumer) string {
```
Comment (georgeharley) [inline plugin.go]: The YAGNI antennae is telling me that _currently_ the story is asking for the second argument to just be the URI of the Git repo to be added. Just sayin' 😉

**Comment by spikymonkey** [comment]
Oh heck - thanks for catching both of those @georgeharley ! Fixed now.

**Comment by spikymonkey** [comment]
Thanks @georgeharley !

**Pull request merged**